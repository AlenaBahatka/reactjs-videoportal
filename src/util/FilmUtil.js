// function to map service structure of response to the correct one
export const remapFilmsStructure = (films = []) => {
	let filmsWithUpdatedStructure = [];
	films.forEach((film) => {
		filmsWithUpdatedStructure.push(remapFilmStructure(film));
	});
	return filmsWithUpdatedStructure;
};

export const remapFilmStructure = (film = {}) => {
	let filmDate = new Date(film.release_date);
	return {
		title: film.title,
		director: film.director || 'Unknown director',
		genre: film.genres && film.genres[0],
		year: filmDate.getFullYear(),
		coverPicture: film.poster_path,
		rating: film.vote_average,
		description: film.tagline,
		fullDescription: film.overview,
		duration: film.runtime,
		id: film.id
	};
};
