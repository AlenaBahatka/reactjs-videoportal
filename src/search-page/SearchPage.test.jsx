import React from 'react';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import 'isomorphic-fetch';
import { Provider } from 'react-redux';
import * as Util from '../util/FilmUtil';
import ConnectedSearchPage, { SearchPage } from './SearchPage';
import thunk from 'redux-thunk';

const film1 = {
	title: 'Roma',
	director: 'Alfonso Cuarón',
	genre: 'comedy',
	year: 2017,
	coverPicture:
		'https://m.media-amazon.com/images/M/MV5BMTU0OTc3ODk4Ml5BMl5BanBnXkFtZTgwMzM4NzI5NjM@._V1_UX182_CR0,0,182,268_AL_.jpg',
	rating: 8,
	description: 'dddd',
	fullDescription: 'ddddddd',
	duration: 124,
	id: 'id1'
};
const film2 = {
	title: 'Tristana',
	director: ' Luis Buñuel',
	genre: 'drama',
	year: 1970,
	coverPicture:
		'https://m.media-amazon.com/images/M/MV5BMDU5ZWE3M2MtZTVhYS00Yjk2LWFkYjQtZDdkNzc1YmJmMGIxXkEyXkFqcGdeQXVyMjUyNDk2ODc@._V1_UY268_CR9,0,182,268_AL_.jpg',
	rating: 9,
	description: 'ffff',
	fullDescription: 'ffffff',
	duration: 123,
	id: 'id2'
};
const film3 = {
	title: 'Roma2',
	director: 'Alfonso Cuarón',
	genre: 'comedy',
	year: 2017,
	coverPicture:
		'https://m.media-amazon.com/images/M/MV5BMTU0OTc3ODk4Ml5BMl5BanBnXkFtZTgwMzM4NzI5NjM@._V1_UX182_CR0,0,182,268_AL_.jpg',
	rating: 7,
	description: 'dddd',
	fullDescription: 'ddddddd',
	duration: 124,
	id: 'id3'
};
let filmsStub = [ film1, film2, film3 ];

let initialSateStub = {
	filterOptions: {
		filterOptions: [ { name: 'title' }, { name: 'genre' } ],
		defaultChecked: 'title'
	},
	searchedFilms: filmsStub,
	sortTypes: [
		{
			id: 'releaseDate',
			displayName: 'release date'
		},
		{
			id: 'rating',
			displayName: 'rating'
		}
	],
	selectedSortType: 'rating'
};
const mockStore = configureStore([ thunk ]);
let store;

describe('SearchPage', () => {
	let fetchFilmsMock;
	beforeEach(() => {
		store = mockStore(initialSateStub);
		fetchFilmsMock = jest.fn();
	});

	it('should correctly render component', () => {
		let searchPageComponent = shallow(<SearchPage searchedFilms={filmsStub} fetchFilms={fetchFilmsMock} />);

		expect(searchPageComponent).toMatchSnapshot();
	});

	describe('handleSearch', () => {
		let filterFilmsMock, sortHandlerMock;
		beforeEach(() => {
			filterFilmsMock = jest.fn();
			sortHandlerMock = jest.fn();
			let searchPageComponent = mount(
				<SearchPage
					searchedFilms={filmsStub}
					filterOptions={initialSateStub.filterOptions}
					filterFilms={filterFilmsMock}
					fetchFilms={fetchFilmsMock}
					sortTypes={initialSateStub.sortTypes}
					sortFilms={sortHandlerMock}
				/>
			);
			searchPageComponent.find(SearchPage).instance().handleSearchClick('Roma2', 'genre');
		});
		it('should call fetching films twice', () => {
			expect(fetchFilmsMock.mock.calls.length).toBe(2);
		});
		it('should call filtering once', () => {
			expect(filterFilmsMock.mock.calls.length).toBe(1);
		});
	});
});
