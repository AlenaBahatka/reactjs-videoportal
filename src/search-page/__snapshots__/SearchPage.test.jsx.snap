// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage should correctly render component 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    <NetflixLabel />
    <SearchHeader
      handleSearchClick={[Function]}
    />
  </div>
  <div
    className="panel-body"
  >
    <SearchToolbar
      numberOfFilms={3}
    />
    <FilmList
      films={
        Array [
          Object {
            "coverPicture": "https://m.media-amazon.com/images/M/MV5BMTU0OTc3ODk4Ml5BMl5BanBnXkFtZTgwMzM4NzI5NjM@._V1_UX182_CR0,0,182,268_AL_.jpg",
            "description": "dddd",
            "director": "Alfonso Cuarón",
            "duration": 124,
            "fullDescription": "ddddddd",
            "genre": "comedy",
            "id": "id1",
            "rating": 8,
            "title": "Roma",
            "year": 2017,
          },
          Object {
            "coverPicture": "https://m.media-amazon.com/images/M/MV5BMDU5ZWE3M2MtZTVhYS00Yjk2LWFkYjQtZDdkNzc1YmJmMGIxXkEyXkFqcGdeQXVyMjUyNDk2ODc@._V1_UY268_CR9,0,182,268_AL_.jpg",
            "description": "ffff",
            "director": " Luis Buñuel",
            "duration": 123,
            "fullDescription": "ffffff",
            "genre": "drama",
            "id": "id2",
            "rating": 9,
            "title": "Tristana",
            "year": 1970,
          },
          Object {
            "coverPicture": "https://m.media-amazon.com/images/M/MV5BMTU0OTc3ODk4Ml5BMl5BanBnXkFtZTgwMzM4NzI5NjM@._V1_UX182_CR0,0,182,268_AL_.jpg",
            "description": "dddd",
            "director": "Alfonso Cuarón",
            "duration": 124,
            "fullDescription": "ddddddd",
            "genre": "comedy",
            "id": "id3",
            "rating": 7,
            "title": "Roma2",
            "year": 2017,
          },
        ]
      }
    />
  </div>
  <Footer />
</div>
`;
